packages/vx-pattern/src/patterns/Circles.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import cx from 'classnames';
+import Pattern from './Pattern';
+
+export type PatternCirclesProps = {
+  /** Unique id for the pattern. */
+  id: string;
+  /** Width of the pattern element. */
+  width: number;
+  /** Height of the pattern element. */
+  height: number;
+  /** Radius of the pattern circles. */
+  radius?: number;
+  /** Fill applied to circles. */
+  fill?: string;
+  /** className applied to circles. */
+  className?: string;
+  /** stroke applied to circles. */
+  stroke?: string;
+  /** strokeWidth applied to circles. */
+  strokeWidth?: number | string;
+  /** strokeDasharray applied to circles. */
+  strokeDasharray?: number | string;
+  /** Whether to fill in circles within the pattern gaps to increase pattern density. */
+  complement?: boolean;
+  /** Background color applied behind cirlces. */
+  background?: string;
+};
+
+export default function PatternCircles({
+  id,
+  width,
+  height,
+  radius = 2,
+  fill,
+  stroke,
+  strokeWidth,
+  strokeDasharray,
+  background,
+  complement = false,
+  className,
+}: PatternCirclesProps) {
+  let corners: number[][] | undefined;
```

Comment:
can change `number[][]` to `[number, number][]` to be more explicit that there are always two numbers.

Reply:
some of the `d3` types are strict about this and others are not. will update 👍